persistent-postgresql/Database/Persist/Postgresql/Internal.hs

```diff
@@ -237,110 +235,45 @@ unBinary (PG.Binary x) = x
 newtype PgInterval = PgInterval {getPgInterval :: NominalDiffTime}
     deriving (Eq, Show)
 
-pgIntervalToBs :: PgInterval -> ByteString
-pgIntervalToBs = toStrictByteString . show . getPgInterval
-
 instance PGTF.ToField PgInterval where
-    toField (PgInterval t) = PGTF.toField t
+    toField = PGTF.toField . fromMaybe (error "PgInterval.toField") . pgIntervalToInterval
```

Comment:
This (and `toPersistValue`) should perhaps saturate/clamp the interval rather than throwing an impure exception. For example if you try to insert `PgInterval 9223372036855`, the existing `toPersistValue` method will generate `"9223372036855s"`, which Postgres will complain about: 

``` hs
SqlError
  { sqlState = "22015"
  , sqlExecStatus = FatalError
  , sqlErrorMsg = "interval field value out of range: \"9223372036855s\""
  , sqlErrorDetail = ""
  , sqlErrorHint = ""
  }
```

Reply:
I understand that the current behaviour is not ideal, but is it necessary to change the `To/FromField` instances for `PgInterval` here? I'd prefer that we leave it as-is if we can (especially if the aim is for downstream users to stop using `PgInterval`), since it's possible that there are downstream users who are relying on the error being thrown at the database level in this case.

Reply:
Unfortunately I think it is necessary to change the instance. The `builtinGetters` are type directed, so I'm forced to decode a PostgreSQL `interval` into a Haskell `Interval`. I could introduce a type that's essentially `Either PgInterval Interval`, but that feels like it's not worth the effort. I can't imagine that someone is relying on the behavior of `PgInterval` throwing a `SqlError` when it's too big.